stats/stats.go

```diff
@@ -0,0 +1,74 @@
+package stats
+
+import (
+	"sync"
+	"time"
+)
+
+type Stats struct {
+	// todo atomic instead of lock?
+	lock *sync.Mutex
+
+	round    uint64
+	sequence uint64
+
+	msgCount      map[string]uint64
+	stateDuration map[string]time.Duration
+}
+
+func NewStats() *Stats {
+	return &Stats{
+		lock:          &sync.Mutex{},
+		msgCount:      make(map[string]uint64),
+		stateDuration: make(map[string]time.Duration),
+	}
+}
+
+func (s *Stats) SetView(sequence uint64, round uint64) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	s.sequence = sequence
+	s.round = round
+}
+
+func (s *Stats) IncrMsgCount(msgType string) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	s.msgCount[msgType]++
+}
+
+func (s *Stats) StateDuration(msgType string, t time.Time) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	s.stateDuration[msgType] = time.Since(t)
+}
```

Comment:
```suggestion
func (s *Stats) StateDuration(state string, t time.Time) {
	s.lock.Lock()
	defer s.lock.Unlock()
	s.stateDuration[state] = time.Since(t)
}
```